### PR TITLE
Reverse and replace

### PR DESCRIPTION
**Pull request by maryrsmcphee**
It works! 

I'm going to make all my testing more robust later... this is just to check that it works

### CODE REVIEW

File: Lab2/lab2_drones_manager.cpp [hunk 1]
```diff
@@ -38,10 +38,10 @@ bool DronesManager::empty() const {
  * @return DroneRecord
  */
 DronesManager::DroneRecord DronesManager::select(unsigned int index) const {
-    if (index > size || index < 0) {
+    if (index >= size || index < 0) {
         cout << "Unable to select: Index is outside of bounds" << endl;
     } else if (empty()) {
-        DroneRecord(0);
+        cout << "Unable to select: List is empty" << endl;
```
Comment (SammyRobensParadise) [inline Lab2/lab2_drones_manager.cpp:45]: I am pretty sure we need a return statement here?

File: Lab2/lab2_drones_manager.cpp [hunk 2]
```diff
@@ -293,7 +293,21 @@ bool DronesManager::remove_back() {
  * @return bool
  */
 bool DronesManager::replace(unsigned int index, DroneRecord value) {
-    return false;
```
Comment (SammyRobensParadise) [inline Lab2/lab2_drones_manager.cpp:296]: can we make sure that replace is just supposed to replace the values and the implementation they are looking for isnt to replace the whole node?

File: Lab2/lab2_drones_manager.cpp [hunk 3]
```diff
@@ -302,7 +316,18 @@ bool DronesManager::replace(unsigned int index, DroneRecord value) {
  * @return bool
  */
 bool DronesManager::reverse_list() {
-    return false;
+    DroneRecord *reversed = this->first;
```
Comment (SammyRobensParadise) [inline Lab2/lab2_drones_manager.cpp]: add a case to see if the list only has one node, and if it does  just return that node

File: Lab2/lab2_drones_manager.cpp (hunk 2 above)
Comment (maryrsmcphee) [reply Lab2/lab2_drones_manager.cpp:296]: Instructions say this, so I assume that it's okay, they specifically say value. 
> PURPOSE: Replaces value at the given index with the given value; the list is not sorted// if the index is invalid, replacement is rejected

File: Lab2/lab2_drones_manager.cpp (hunk 3 above)
Comment (maryrsmcphee) [reply Lab2/lab2_drones_manager.cpp]: it just returns a bool, so it'll just return true.

**Comment by maryrsmcphee** [comment]
Actually wait a minute to review I think reverse_list() could be better

**Review by SammyRobensParadise — APPROVED** [review]

**Pull request merged**